lang/csharp/src/apache/test/File/FileTests.cs

```diff
@@ -198,6 +199,45 @@ public void TestGenericData(string schemaStr, object[] value, Codec.Type codecTy
             }
         }
 
+        /// <summary>
+        /// This test is a single test case of the previous test but introduce a DeflateStream
+        /// as it is a standard non-seekable Stream that have the same behavior as the
+        /// NetworkStream that should be handled.
+        /// </summary>
+        [TestCase("{\"type\":\"record\", \"name\":\"n\", \"fields\":" +
+            "[{\"name\":\"f1\", \"type\":[\"int\", \"long\"]}]}",
+            new object[] { "f1", 100L }, Codec.Type.Null)]
+        public void TestNonSeekableStream(string schemaStr, object[] value, Codec.Type codecType)
+        {
+            foreach (var rwFactory in GenericOptions<GenericRecord>())
+            {
+                // Create and write out
+                MemoryStream compressedStream = new MemoryStream();
+                // using here a DeflateStream as it is a standard non-seekable stream, so if it works for this one,
+                // it should also works with any standard non-seekable stream (ie: NetworkStreams)
+                DeflateStream dataFileOutputStream = new DeflateStream(compressedStream, CompressionMode.Compress);
+                using (var writer = rwFactory.CreateWriter(dataFileOutputStream, Schema.Parse(schemaStr), Codec.CreateCodec(codecType)))
+                {
+                    writer.Append(mkRecord(value, Schema.Parse(schemaStr) as RecordSchema));
+                }
```

Comment:
Ensure we throw an exception when calling `Sync` with a non-seekable stream.

```suggestion

                    // The Sync method is not supported for non-seekable streams.
                    Assert.Throws<NotSupportedException>(() => writer.Sync());
                }
```